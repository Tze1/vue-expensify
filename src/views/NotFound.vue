<template>
  <div class="notfound component">
    <h2>Not Found</h2>
    <p>Sorry!  The resource you requested was not found.</p>
    <p>Please go back to:</p>
    <router-link v-if="!userLoggedIn" to="/">Login</router-link>
    <router-link v-else to="/home">Home</router-link>
  </div>
</template>

<script>
  import { mapState } from 'vuex';

  export default {
    computed: mapState({
      userLoggedIn () {
        return this.$store.getters.userLoggedIn;
      },
    }),
  };
</script>

<style lang="scss">
  @import '../styles/app.scss';
</style>
