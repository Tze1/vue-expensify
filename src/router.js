import Vue from 'vue'
import Router from 'vue-router'
import Login from './views/Login.vue'
import Dashboard from './views/Dashboard'
import Home from './views/Home.vue'
import NotFound from './views/NotFound';

Vue.use(Router)

// IMPORTANT: ALL secure routes must have meta: { requiresAuth: true }
export default new Router({
  mode: 'history',
  base: process.env.BASE_URL,
  routes: [
    {
      path: '/',
      name: 'Login',
      component: Login,
    },
    {
      path: '/home',
      name: 'Home',
      component: Home,
      meta: {
        requiresAuth: true,
      },
    },
    {
      path: '/dashboard',
      name: 'Dashboard',
      component: Dashboard,
      meta: {
        requiresAuth: true,
      },
    },
    {
      path: '/about',
      name: 'about',
      // route level code-splitting
      // this generates a separate chunk (about.[hash].js) for this route
      // which is lazy-loaded when the route is visited.
      component: () => import(/* webpackChunkName: "about" */ './views/About.vue')
    },
    {
      path: '*',
      name: 'NotFound',
      component: NotFound
    }
  ],
});
