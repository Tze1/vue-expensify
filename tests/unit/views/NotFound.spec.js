import { shallowMount } from '@vue/test-utils'
import router from '@/router'
import NotFound from '@/views/NotFound'

describe('NotFound.vue', () => {
  it('renders NotFound view properly', () => {
    const $store = {
      state: {
        auth: {
          user: {
            displayName: 'Test',
            photoUrl: '/img/avatar-default.png',
          },
        },
      },
      getters: {
        userLoggedIn: state => {
          return (state.user !== undefined);
        },
      },
    };
    const wrapper = shallowMount(NotFound, {
      mocks: {
        $store,
      },
    });

    expect(wrapper).toMatchSnapshot();
  })
})
