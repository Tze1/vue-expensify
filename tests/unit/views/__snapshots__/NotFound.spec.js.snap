// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound.vue renders NotFound view properly 1`] = `
<div class="notfound component">
  <h2>Not Found</h2>
  <p>Sorry! The resource you requested was not found.</p>
  <p>Please go back to:</p>
  <router-link-stub to="/home" tag="a" event="click">Home</router-link-stub>
</div>
`;
